### ISSUE TITLE
Container writes to `/data`; needs write access.

### ISSUE THREAD

**Issue by Myridium** [christophetd/duplicacy-autobackup#34]
This docker container relies on the `/data` mountpoint being writable. This doesn't make sense, as in principle (and for safety reasons) duplicacy should only need read-access to the data it is backing up, and should not modify that data. Mounting `/data` as read-only after `.duplicity` is created causes this docker container to first claim that the directory is already initialised, then complain  that it isn't initialised as `[ ! -d .duplicity]` fails (I do not know why).

This seems to be a design flaw of Duplicacy which assumes the user is happy to create a directory `.duplicacy` in the current working directory containing metadata about the duplicacy backup. Now what is the point of such a thing? If you're backing up, you shouldn't have to back up metadata associated with the backup. How do I backup the metadata of the metadata? Et cetera, ad infinitum.

I propose having this container change directory to a dedicated working directory, e.g. `/wd` before beginning backup. I don't know why the `.duplicity` directory is needed at all. But if the user insists, then `/wd` can be mounted as a volume (persistent storage), which the user does not backup.

**Comment by christophetd** [comment]
That's a very good point. Would it be worth asking the Duplicacy maintainers guidance on that? Or have you tested that having the .duplicacy directory in a different folder works as expected?

**Comment by Myridium** [comment]
I gave up on Duplicacy as I could not understand their system. I could not understand whether this '.duplicacy' directory was actually necessary or not. I could not restore a repo without it, so I guess it is. I'm actually not sure how I'm supposed to recover backups with this Docker container, because don't I need the `.duplicacy` directory to restore? The one that's only stored with the data being backed up...? Is this nonsensical or is there a way to restore without .duplicacy?

The use of a hidden directory `.duplicacy` is poor design because hidden files can be missed by various software (including shell globbing for example). Backup software should be designed to prevent all data loss, including through user error. I could not understand how this Duplicacy system is supposed to work, hence I gave up on it. It seems geared toward cloud backups anyway. I'd be willing to give another try at some point.

As for addressing the read-only issue, I did some internet sleuthing and found out that duplicacy will follow symlinks in the working directory where it backs things up, according to the creator on a forum post some years ago (I didn't save the link). 

As for this docker: I think this would work for allowing backup read-only backup of `/data`: change working directory to `/wd` as the working directory. Create a symlink `/wd/data --> /data`. Then backup to working directory, and copy contents of `/wd` to output.

Or, maybe change directory to the location of the backup output. And create the symlink there.
